### PR TITLE
Fix web3 and ethers types

### PR DESCRIPTION
**Pull request by germartinez**
## What it solves
Resolves https://github.com/gnosis/safe-core-sdk/issues/138

## How this PR fixes it
- Removes `any` type from `web3` and `ethers` variables
- Removes `AbiItem` local type
- Fixes tests by removing the `deterministicDeployment` property from `ERC20Mintable` contract deployment (hardhat)
- Updates deps

### CODE REVIEW

File: packages/safe-core-sdk/tests/utils/setupEthAdapter.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export async function getEthAdapter(signer: Signer): Promise<EthAdapter> {
   switch (process.env.ETH_LIB) {
     case 'web3':
       const signerAddress = await signer.getAddress()
-      const web3AdapterConfig: Web3AdapterConfig = { web3, signerAddress }
+      const web3AdapterConfig: Web3AdapterConfig = { web3: web3 as any, signerAddress }
```
Comment (germartinez) [inline packages/safe-core-sdk/tests/utils/setupEthAdapter.ts:12]: This is needed. `web3` instance is coming from `hardhat` package (not `web3`) and tests would complain telling that "is not assignable".

File: packages/safe-core-sdk-types/src/contracts/GnosisSafeContract.ts [hunk 2]
```diff
@@ -22,6 +22,6 @@ export interface GnosisSafeContract {
     safeTransaction: SafeTransaction,
     options?: TransactionOptions
   ): Promise<TransactionResult>
-  encode(methodName: any, params: any): string
+  encode(methodName: string, params: any): string
```
Comment (iamacook) [inline packages/safe-core-sdk-types/src/contracts/GnosisSafeContract.ts:25]: Can you type this `any`?

File: packages/safe-web3-lib/src/Web3Adapter.ts [hunk 3]
```diff
@@ -99,15 +98,15 @@ class Web3Adapter implements EthAdapter {
     return getGnosisSafeProxyFactoryContractInstance(safeVersion, proxyFactoryContract)
   }
 
-  getContract(address: string, abi: AbiItem[]): any {
-    return new this.#web3.eth.Contract(abi, address)
+  getContract(address: string, abi: AbiItem | AbiItem[], options?: ContractOptions): any {
```
Comment (iamacook) [inline packages/safe-web3-lib/src/Web3Adapter.ts:101]: Does this really return `any`? 👀

File: packages/safe-web3-lib/src/Web3Adapter.ts [hunk 4]
```diff
@@ -119,7 +118,7 @@ class Web3Adapter implements EthAdapter {
     return this.#web3.eth.sign(message, this.#signerAddress)
   }
 
-  estimateGas(transaction: EthAdapterTransaction, options?: string): Promise<number> {
+  estimateGas(transaction: EthAdapterTransaction, options?: any): Promise<number> {
```
Comment (iamacook) [inline packages/safe-web3-lib/src/Web3Adapter.ts:121]: `any` options?

File: packages/safe-core-sdk-types/src/contracts/GnosisSafeContract.ts [hunk 5]
```diff
@@ -22,6 +22,6 @@ export interface GnosisSafeContract {
...
   estimateGas(methodName: string, params: any[], options: TransactionOptions): Promise<number>
```
Comment (iamacook) [inline packages/safe-core-sdk-types/src/contracts/GnosisSafeContract.ts:26]: And these `params`?

**Review by iamacook — COMMENTED** [review]
`any` is not your friend 🤣

File: packages/safe-core-sdk-types/src/contracts/GnosisSafeContract.ts (hunk 2 above)
Comment (iamacook) [reply packages/safe-core-sdk-types/src/contracts/GnosisSafeContract.ts:26]: After discussing this, I'm happy to resolve this with `any`. You may be able to strictly type it via using generics and linking `methodName` to a predefined type of strings. When `T extends <methodNameHere>` you can use a ternary to select the `params`. It's complex so I'll leave it up to you if you want to include it.

File: packages/safe-core-sdk-types/src/contracts/GnosisSafeContract.ts (hunk 5 above)
Comment (iamacook) [reply packages/safe-core-sdk-types/src/contracts/GnosisSafeContract.ts:27]: I'm resolving this after discussing it. Please see my comment above for the reasoning.

File: packages/safe-web3-lib/src/Web3Adapter.ts (hunk 3 above)
Comment (iamacook) [reply packages/safe-web3-lib/src/Web3Adapter.ts:101]: It seems like the type is `'TODO:'` on web3's end. It's possible to use `ReturnType<typeof this.#web3.eth.Contract>` to infer it I think. I'll resolve this, however, as it is not properly typed by a third party.

**Review by iamacook — APPROVED** [review]
Thanks for discussing the remaining issues with me. Looks great!

**Pull request merged**